### PR TITLE
fix: handle abbreviations and numbers in feature name

### PR DESCRIPTION
**Pull request by lanej**

### CODE REVIEW

File: lib/toggles/constant_lookup.rb [hunk 1]
```diff
@@ -8,35 +8,42 @@ def initialize(sym)
...
-          )
+          # translate class name into path part i.e :BearDog #=> 'bear_dog'
+          key =
+            sym
+            .to_s
+            .gsub(/([A-Z]{2,})/) { |s| s.chars.join('_') }
+            .gsub(/([a-z])([A-Z])/) { |s| s.chars.join('_') }
+            .downcase
+            .gsub(/([a-z])([0-9])/) { |s| s.chars.join('_') }
+            .to_sym
```
Comment (kgraves) [inline lib/toggles/constant_lookup.rb:36]: This doesn't look like it'll work for strings with a number directly before a capital letter, e.g. `"Allow2DayService"`, it translates to `:allow_2day_service`, I'm thinking it should be `:allow_2_day_service` based on the CN22 example, right? Unless you didn't want the separation after the number? It looks like we'll just need another `gsub` with the converse, digit before letter, right?

Comment (lanej) [reply lib/toggles/constant_lookup.rb:36]: hmm, ill add another test case.

File: lib/toggles/constant_lookup.rb [hunk 2]
```diff
@@ -8,35 +8,38 @@ def initialize(sym)
...
         def const_missing(sym)
-          subtree_or_feature = features.fetch(
-            # translate class name into path part i.e :BearDog #=> 'bear_dog'
-            sym.to_s.gsub(/([a-z])([A-Z])/) { |s| s.chars.join('_') }.downcase.to_sym,
-          )
+          # translate class name into path part i.e :BearDog #=> 'bear_dog'
+          key = sym.to_s
+          key.gsub!(/([A-Z\d]+)([A-Z][a-z])/, '\1_\2'.freeze)
+          key.gsub!(/([a-z\d])([A-Z])/, '\1_\2'.freeze)
+          key.downcase!
```
Comment (kgraves) [inline lib/toggles/constant_lookup.rb:32]: I'm still seeing that we're having some issues with translation of potential keys, but let me know if you don't see it as an issue right now, or something we'll fix later.

```
[2] pry(main)> def tr(str)
[2] pry(main)*   str.gsub!(/([A-Z\d]+)([A-Z][a-z])/, '\1_\2'.freeze)
[2] pry(main)*   str.gsub!(/([a-z\d])([A-Z])/, '\1_\2'.freeze)
[2] pry(main)*   str.downcase!
[2] pry(main)*   return str
[2] pry(main)* end
=> :tr
[3] pry(main)>
[4] pry(main)> tr("CN22")
=> "cn22"
[5] pry(main)> tr("Allow2DayAir")
=> "allow2_day_air"
[6] pry(main)> tr("Allow2ndDaySomething")
=> "allow2nd_day_something"
```

Comment (lanej) [reply lib/toggles/constant_lookup.rb:32]: As this a transitionary release and there is a workaround for it, I'm inclined to improve it within reason and then move on to a major release that removes support for module-based access entirely.

Comment (kgraves) [reply lib/toggles/constant_lookup.rb:32]: cool, works for me

**Review by kgraves — APPROVED** [review]

**Pull request merged**